enterprise/api/v1/views.py

```diff
@@ -200,6 +210,93 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @detail_route(methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    def enterprise_learners(self, request, pk):
+        """
+        Creates a set of enterprise_learners by enrolling them in the specified course.
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        if serializer.is_valid():
+            singular_email = serializer.data.get('email')
+            emails = set()
+            already_linked_emails = []
+            duplicate_emails = []
+            errors = []
+            if singular_email:
+                emails.add(singular_email)
+            try:
+                for email in emails:
+                    try:
+                        already_linked = validate_email_to_link(email, ignore_existing=True)
+                    except ValidationError as error:
+                        errors.append(error)
+                    else:
+                        if already_linked:
+                            already_linked_emails.append((email, already_linked.enterprise_customer))
+                        elif email in emails:
+                            duplicate_emails.append(email)
+                        else:
+                            emails.add(email)
+            except ValidationError as exc:
+                errors.append(exc)
+
+            if errors:
+                return Response(errors, status=HTTP_400_BAD_REQUEST)
+
+            for email in emails:
+                models.EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+
+            course_run_key = serializer.data.get('course_run_key')
+            mode = serializer.data.get('course_mode')
+            if course_run_key:
+                this_customer_linked_emails = [
+                    email for email, customer in already_linked_emails if customer == enterprise_customer
+                ]
+                linked_learners = list(emails) + this_customer_linked_emails
+                if linked_learners:
```

Comment:
Same as above, if there are no linked learners, should we just return 202, or do something else?

Reply:
It still accepts the payload so I think a 202 is still appropriate here even if it ends up being a no-op, one of the downsides of taking a form and trying to pretend it's a REST API now.